### PR TITLE
Update skeleton

### PR DESCRIPTION
**Pull request by aawdls**
Apply the latest skeleton repo.

Ditch changes to the docs as current build is fine.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (AlexanderWells-diamond) [inline README.rst:1]: Improperly completed merge?

Comment (aawdls) [reply README.rst:1]: Good spot thank you

**Comment by aawdls** [comment]
I also need to re-enable the code CI

**Comment by aawdls** [comment]
The docs build is failing with 

```
  File "/opt/hostedtoolcache/Python/3.11.3/x64/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for coniql
```

but it does not do this locally so I will compare what I am doing to the docs CI.

**Comment by AlexanderWells-diamond** [comment]
I've fixed the CI, as well as a few miscellaneous formatting issues and black/mypy issues. I think this is ready.

Note that we still need to rename master -> main, and migrate the whole repo to DiamondLightSource.

**Pull request merged**